x/evm/querier.go

```diff
@@ -58,27 +56,22 @@ func queryProtocolVersion(keeper Keeper) ([]byte, sdk.Error) {
 func queryBalance(ctx sdk.Context, path []string, keeper Keeper) ([]byte, sdk.Error) {
 	addr := ethcmn.BytesToAddress([]byte(path[1]))
 	balance := keeper.GetBalance(ctx, addr)
-	hBalance := &hexutil.Big{}
-	err := hBalance.UnmarshalText(balance.Bytes())
-	if err != nil {
-		panic("could not marshal big.Int to hexutil.Big")
-	}
-
-	bRes := types.QueryResBalance{Balance: hBalance}
-	res, err := codec.MarshalJSONIndent(keeper.cdc, bRes)
+	res, err := codec.MarshalJSONIndent(keeper.cdc, balance)
 	if err != nil {
-		panic("could not marshal result to JSON")
+		panic("could not marshal result to JSON: ")
 	}
 
 	return res, nil
 }
 
 func queryBlockNumber(ctx sdk.Context, keeper Keeper) ([]byte, sdk.Error) {
 	num := ctx.BlockHeight()
-	bnRes := types.QueryResBlockNumber{Number: big.NewInt(num)}
-	res, err := codec.MarshalJSONIndent(keeper.cdc, bnRes)
+	hexBig := hexutil.Uint64(num)
```

Comment:
Small nitpick but confusing name since it's a hexutil uint. Also, the type that is being received on the RPC api is just a uint64, reason for marshalling it as this type?

Reply:
I see `hexutil.Uint64` all the way through: https://github.com/ChainSafe/ethermint/blob/9812df303517a9324710585cbabbba196f066bd7/rpc/eth_api.go#L91

Reply:
ah, my bad actually (was looking at old type I think), if you do another commit though to change hexBig to hexUint or something more accurate would be great, but fine with this